modules/EvManager/tests/CMakeLists.txt

```diff
@@ -1,9 +1,16 @@
-set(TARGET_NAME test_${PROJECT_NAME}_EvManager)
-add_executable(${TARGET_NAME})
+set(TEST_TARGET_NAME ${PROJECT_NAME}_EvManager_tests)
+add_executable(${TEST_TARGET_NAME})
 
 add_dependencies(${TEST_TARGET_NAME} ${MODULE_NAME})
 
-target_sources(${TARGET_NAME}
+get_target_property(GENERATED_INCLUDE_DIR generate_cpp_files EVEREST_GENERATED_INCLUDE_DIR)
+target_include_directories(${TEST_TARGET_NAME} PRIVATE
+    . ..
+    ${GENERATED_INCLUDE_DIR}
+    ${CMAKE_BINARY_DIR}/generated/modules/${MODULE_NAME}
+)
+
```

Comment:
```
target_include_directories(${TEST_TARGET_NAME}
    PRIVATE
        "$<TARGET_PROPERTY:generate_cpp_files,EVEREST_GENERATED_INCLUDE_DIR>"
)
```
might be shorter.  Also note, that you're depending here on files generated by the target `generate_cpp_files` - as cmake doesn't know, that you're depending on this target, it could happen, that the test gets build before the files are generated (although I haven't seen that happening yet).
Furthermore, including the `.` in `target_include_directories` doesn't make sense to me, if the target itself is located in that directory.

Reply:
Yes this solution is nicer, thanks!